File: QadeerApp.Web/Modules/ServerTypes/Administration/RoleRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface RoleRow {
    RoleId?: number;
    RoleName?: string;
}

export abstract class RoleRow {
    static readonly idProperty = 'RoleId';
    static readonly nameProperty = 'RoleName';
    static readonly localTextPrefix = 'Administration.Role';
    static readonly lookupKey = 'Administration.Role';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<RoleRow>('Administration.Role') }
    static async getLookupAsync() { return getLookupAsync<RoleRow>('Administration.Role') }

    static readonly deletePermission = 'Administration:Security:Roles:Delete';
    static readonly insertPermission = 'Administration:Security:Roles:Insert';
    static readonly readPermission = 'Administration:Security:Roles:View';
    static readonly updatePermission = 'Administration:Security:Roles:Update';

    static readonly Fields = fieldsProxy<RoleRow>();
}

File: QadeerApp.Web/Modules/Administration/Role/RoleRow.cs
namespace QadeerApp.Administration;

[ConnectionKey("Default"), Module("Administration"), TableName("Roles")]
[DisplayName("Roles"), InstanceName("Role")]
[ReadPermission(PermissionKeys.Role.View)]
[ModifyPermission(PermissionKeys.Role.Update)]
[InsertPermission(PermissionKeys.Role.Insert)]
[UpdatePermission(PermissionKeys.Role.Update)]
[DeletePermission(PermissionKeys.Role.Delete)]
[LookupScript]
public sealed class RoleRow : Row<RoleRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Role Id"), Identity, ForeignKey("Roles", "RoleId"), LeftJoin("jRole"), IdProperty]
    public int? RoleId { get => fields.RoleId[this]; set => fields.RoleId[this] = value; }

    [DisplayName("Role Name"), Size(100), NotNull, QuickSearch, NameProperty]
    public string RoleName { get => fields.RoleName[this]; set => fields.RoleName[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field RoleId;
        public StringField RoleName;
    }
}
